app/forms/candidate_interface/volunteering_role_form.rb

```diff
@@ -3,16 +3,38 @@ class VolunteeringRoleForm
     include ActiveModel::Model
     include DateValidationHelper
 
-    attr_accessor :id, :role, :organisation, :details, :working_with_children,
-                  :start_date_day, :start_date_month, :start_date_year,
-                  :end_date_day, :end_date_month, :end_date_year
-
-    validates :role, :organisation, :working_with_children, presence: true
+    attr_accessor :id,
+                  :role,
+                  :organisation,
+                  :details,
+                  :working_with_children,
+                  :start_date_day,
+                  :start_date_month,
+                  :start_date_year,
+                  :start_date_unknown,
+                  :currently_working,
+                  :end_date_day,
+                  :end_date_month,
+                  :end_date_year,
+                  :end_date_unknown
+
+    validates :role,
+              :organisation,
+              :working_with_children,
+              presence: true
 
     validates :role, :organisation, length: { maximum: 60 }
-
     validates :start_date, date: { presence: true, future: true, month_and_year: true, before: :end_date }
-    validates :end_date, date: { future: true, month_and_year: true }, unless: :start_date_blank?
+
+    with_options if: :restructured_work_history_flag_active? do |form|
```

Comment:
Nice. Was not aware you could group validations like this

Reply:
Neither did I. But I spent a lot of time reading this page https://guides.rubyonrails.org/active_record_validations.html#grouping-conditional-validations

I think it improves the readability and makes it clear which context those validations should be executed in.

Reply:
Yep absolutely